include/boost/url/rfc/impl/query_rule.ipp

```diff
@@ -64,13 +64,6 @@ parse(
                 BOOST_URL_RETURN_EC(
                     error::bad_pct_hexdig);
             }
-            if (it[1] == '0' &&
-                it[2] == '0')
-            {
-                // null in input
-                BOOST_URL_RETURN_EC(
-                    error::illegal_null);
```

Comment:
Are nulls allowed in `application/x-www-form-urlencoded`?

Reply:
I don't know. I would guess it's valid in certain cases. But the problem, in this case, is that query_rule is not being used for `application/x-www-form-urlencoded` only. It's used for queries in general and `url`s with nulls cannot be parsed unless `query_rule` accepts nulls. Organizing the query into key/values is also common in schemes other than http.

Reply:
> I don't know.

Well what does the spec say

Reply:
The URI spec says it's OK. The HTTP spec doesn't say anything.